supervisor/__main__.py

```diff
@@ -55,7 +55,7 @@ def run_os_startup_check_cleanup() -> None:
     loop.run_until_complete(coresys.core.connect())
 
     bootstrap.supervisor_debugger(coresys)
-    bootstrap.migrate_system_env(coresys)
+    loop.run_until_complete(bootstrap.migrate_system_env(coresys))
```

Comment:
This feels a bit hacky.

That said, there is more I/O being done in `migrate_system_env`. But `migrate_system_env` seems to be old code anyways, even the comment says we can remove it in 2022. So let's just drop that function entirely, so we don't have to touch this.